File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.go
func minAddToMakeValid(s string) int {
    arr:=[]rune{}
    Imbalance:=0
    for _,c:= range s{
        if c=='('{
            arr=append(arr,c)
        }else if len(arr)>0{
            arr=arr[:len(arr)-1]
        }else{
            Imbalance++
        }
    }
    return len(arr)+Imbalance
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int OpenCount = 0;
        int Imbalance = 0;
        for(char c:s){
            if(c=='('){
                OpenCount++;
            }else if(OpenCount){
                OpenCount--;
            }else{
                Imbalance++;
            }
        }        
        return Imbalance + OpenCount;
    }
};